xml/fingerprints.xsd

```diff
@@ -7,6 +7,9 @@
       <xsd:element name="fingerprint" type="fingerprint_element" minOccurs="1" maxOccurs="unbounded"/>
     </xsd:sequence>
     <xsd:attribute name="matches" type="xsd:string" use="optional"/>
+    <xsd:attribute name="protocol" type="xsd:string" use="optional"/>
```

Comment:
It would be good to add some comments about these fields so that future readers/users of this schema don't need to read the code to understand these values and their uses.  Either use XML comments or the xsd `annotation`, which I think is more correct here.